### DISCUSSION TITLE
Bad state: Expected exactly one element, but got 0

### DISCUSSION THREAD

**Discussion by givip** [simolus3/drift#1481] [Q&A]
`Bad state: Expected exactly one element, but got 0`
exception produced by using 
`watchSingleOrNull()` or `watchSingle()`
but how is it possible and how to avoid it?
I'm using Moor 4.5.0 with Flutter 2.5.2

**Comment by simolus3** [comment]
Did this really happen with `watchSingleOrNull()`? If so, can you post the stack trace?

When using `watchSingle()` on a result set that may have zero rows, you risk this error. You should use `watchSingleOrNull()` in that case.

**Reply by givip** [reply]
I'll check again.
In this case you should correct `StateError` message, because it has the same as for `watchSingle()`, but in fact, this is not possible.

**Reply by simolus3** [reply]
When the error happens for `watchSingleOrNull()` as well it's likely coming from somewhere else. Either way, a stack trace or some code that reproduces this would be helpful for me to check what's going on here.
